os/kernel/log_dump/log_dump.c

```diff
@@ -184,6 +248,77 @@ int log_dump_save(char ch)
 	return LOG_DUMP_OK;
 }
 
+int log_dump_save(char ch)
+{
+	size_t free_size = 0;
+	int ret = 0;
+
+	uncomp_array[uncomp_curbytes] = ch;
+	uncomp_curbytes++;
+	if (uncomp_curbytes == CONFIG_LOG_DUMP_CHUNK_SIZE) {
+		/* compress the block, add it to the nodes, reset the uncomp_array */
+		set_comp_head = true;
+		unsigned char *out_buf = (unsigned char *)kmm_malloc(CONFIG_LOG_DUMP_CHUNK_SIZE + LZMA_PROPS_SIZE);
```

Comment:
I think it is better to check the result of memory allocation.

Reply:
done